src/dungeon_leader.c

```diff
@@ -0,0 +1,27 @@
+#include "global.h"
+#include "dungeon_leader.h"
+
+#include "dungeon_global_data.h"
+#include "dungeon_util.h"
+
+extern struct DungeonEntity *gLeaderPokemon;
+
+struct DungeonEntity* GetLeaderEntity()
+{
+    struct DungeonEntity *leader = gLeaderPokemon;
+    if (leader == NULL)
+    {
+        s32 i = 0;
```

Comment:
Looks like it also matches with a for loop:
https://staging.cexplore.karathan.at/z/c4Urxy

Reply:
I tried this at one point and it didn't match, but it seems another change I made caused the for loop to work. Updated.